pkg/api/http_server.go

```diff
@@ -131,6 +132,7 @@ type HTTPServer struct {
 	TeamPermissionsService       *resourcepermissions.Service
 	NotificationService          *notifications.NotificationService
 	DatasourcePermissionsService DatasourcePermissionsService
+	dashboardsnapshotsService    dashboardsnapshots.Service
```

Comment:
Think you need to use pointer receiver here since `dashboardsnapshots.Service` is a struct and not an interface.